### PR TITLE
Clarify the new framework based environment variables for plugin discovery

### PR DESCRIPTION
**Pull request by nkolev92**
Related to https://github.com/NuGet/Home/issues/8151 & https://github.com/NuGet/NuGet.Client/pull/2986. 

This should only be merged once the linked PR is completed and 5.3 has been released. 

The specs at https://github.com/NuGet/Home/wiki/NuGet-cross-plat-authentication-plugin and https://github.com/NuGet/Home/wiki/NuGet-Package-Download-Plugin might need updated, although they contain the same information as the actual docs.

### CODE REVIEW

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[69e0198](https://github.com/NuGet/docs.microsoft.com-nuget/commits/69e01987c00936267f333eaf7d685ff47b244047)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/reference/extensibility/NuGet-Cross-Platform-Plugins.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/master-nkolev92-xplatPluginEnvVar/docs/reference/extensibility/NuGet-Cross-Platform-Plugins.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/reference/extensibility/nuget-cross-platform-plugins?branch=pr-en-us-1599) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C8%5C6%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201908062320473744-1599%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=yEhLh0TlTeNb27IDt8kGmv1E%2BQ2NPUjDGfDXe7Arhsw%3D&st=2019-08-06T23%3A16%3A47Z&se=2019-09-06T23%3A21%3A47Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/reference/extensibility/NuGet-Cross-Platform-Plugins.md [hunk 1]
```diff
@@ -66,8 +66,10 @@ After 1 minute of inactivity a plugin is considered idle and is shut down.
 ## Plugin installation and discovery
 
 The plugins will be discovered via a convention based directory structure.
-CI/CD scenarios and power users can use an environment variable to override the behavior.
+CI/CD scenarios and power users can use environment variables to override the behavior. Note that `NUGET_NETFX_PLUGIN_PATHS` and `NUGET_NETCORE_PLUGIN_PATHS` are only available with 5.3+ version of the NuGet tooling and later.
 
+- `NUGET_NETFX_PLUGIN_PATHS` - defines the plugins that will be used by the .NET Framework based tooling (NuGet.exe/MSBuild.exe/Visual Studio). Takes precedence over `NUGET_PLUGIN_PATHS`. (NuGet version 5.3+ only)
+- `NUGET_NETCORE_PLUGIN_PATHS` - defines the plugins that will be used by the .NET Core based tooling (dotnet.exe). Takes precedence over `NUGET_PLUGIN_PATHS`. (NuGet version 5.3+ only)
 - `NUGET_PLUGIN_PATHS` - defines the plugins that will be used for that NuGet process, priority reserved. If this environment variable is set, it overrides the convention based discovery.
```
Comment (zarenner) [inline docs/reference/extensibility/NuGet-Cross-Platform-Plugins.md]: Even though it's hinted at above, perhaps explicitly state that `NUGET_PLUGIN_PATHS` is ignored if one of the framework-specific environment variables is used?

Comment (nkolev92) [reply docs/reference/extensibility/NuGet-Cross-Platform-Plugins.md]: Makes sense. 
Thanks.

**Comment by karann-msft** [comment]
Docs Build status updates of commit _[21c5021](https://github.com/NuGet/docs.microsoft.com-nuget/commits/21c50213d8e5f2a6b20d39d04e032f24c056c186)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/reference/extensibility/NuGet-Cross-Platform-Plugins.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/master-nkolev92-xplatPluginEnvVar/docs/reference/extensibility/NuGet-Cross-Platform-Plugins.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/nuget/reference/extensibility/nuget-cross-platform-plugins?branch=pr-en-us-1599) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C8%5C7%5C19a7edc0-f91d-0fd5-6e5f-4d131db05c11%5CPullRequest%5C201908071808497125-1599%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=NKENByAB8gg6fLogq1i2qZ8QG8LCw7Dj07G371AdvJQ%3D&st=2019-08-07T18%3A04%3A52Z&se=2019-09-07T18%3A09%3A52Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by nkolev92** [comment]
Now that Preview 3 has shipped, can I get a review on this?

**Pull request merged**